### PR TITLE
feat(assignment-5):search functionality to filter out team member

### PR DESCRIPTION
**Pull request by aditifarkya**
**Description**:
Filter out team members on the basis of search text in search bar.
Added sass package to support scss files.

**Functionality After:**
Assignment5-2
assignment5-1
assignment5

### CODE REVIEW

File: src/components/header-component/index.js [hunk 1]
```diff
@@ -12,13 +12,16 @@ const HeaderComponent = () => {
         style={{ height: "100px", marginLeft: "20px", marginRight: "20px" }}
       />
       <h1 style={{ color: "deeppink" }}>Brainly Fools</h1>
-      <div className="header__search">
+      {/* commenting out to move it in separate component */}
```
Comment (ap221882) [inline src/components/header-component/index.js:15]: Well appreciated comment.

File: src/components/search-component/index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { useState } from "react";
```
Comment (ap221882) [inline src/components/search-component/index.js:1]: Leave a blank line between global and local imports. Increases readability.

File: src/components/search-component/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+const SearchComponent = ({ setsearchfiltertext }) => {
+  const [searchval, setsearchval] = useState("");
+  //function to submit search value
+  const submitsearch = (e) => {
+    e.preventDefault();
+    setsearchfiltertext(filtersearchdata(searchval));
+  };
+  // function return arry of filter out data
+  const filtersearchdata = (searchval) => {
+    return teamdata.filter((search) => {
```
Comment (ap221882) [inline src/components/search-component/index.js:13]: Try to make this a utility/reusable function.

File: src/components/search-component/index.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      return (
+        search.name.toLowerCase().includes(searchval.toLowerCase()) ||
+        search.designation.toLowerCase().includes(searchval.toLowerCase())
+      );
+    });
+  };
+  return (
+    <div>
```
Comment (ap221882) [inline src/components/search-component/index.js:21]: No need of div here

File: src/index.js [hunk 5]
```diff
@@ -1,34 +1,48 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
+import { useState } from "react";
```
Comment (ap221882) [inline src/index.js:3]: You can 
```
import React, {useState} from 'react';
```

File: src/index.scss [hunk 6]
```diff
@@ -0,0 +1,48 @@
+body {
+  font-family: sans-serif;
+}
+.rootele {
```
Comment (ap221882) [inline src/index.scss:4]: Can give it a proper name.

File: src/index.js [hunk 7]
```diff
@@ -1,34 +1,48 @@
...
+  // let card = [];
+  // for (let i = 0; i < teamdata.length; i++) {
+  //   console.log(i);
+  //   console.log(teamdata[i]);
+  //   card.push(<CardComponent teamdata={teamdata[i]} key={teamdata.id} />);
+  // }
+  console.log(searchfiltertext);
+  const searchingresult = () => {
+    return searchfiltertext.map((element) => {
+      return <CardComponent teamdata={element} key={element.id} />;
```
Comment (ap221882) [inline src/index.js:37]: If that's for playing around then it's okay. But you should put your component logic inside/after the return keyword only, e.g., here- in return we are calling a function that returns an array of JSX elements. Doesn't one have to think to read it?... If one person needs to think means you should make it more readable.

**Review by ap221882 — COMMENTED** [review]
Nicely written. I liked scss.

File: src/components/search-component/index.js (hunk 3 above)
Comment (aditifarkya) [reply src/components/search-component/index.js:13]: Thank you for comment.
sure, will make it common if any use case will appear in future.

File: src/index.js (hunk 5 above)
Comment (aditifarkya) [reply src/index.js:3]: done

File: src/components/search-component/index.js (hunk 4 above)
Comment (aditifarkya) [reply src/components/search-component/index.js:21]: agree !
kept it to have proper div structure.

**Pull request merged**